Python/instrumentation.c

```diff
@@ -1282,9 +1303,16 @@ initialize_tools(PyCodeObject *code)
     for (int i = 0; i < code_len; i++) {
         _Py_CODEUNIT *instr = &_PyCode_CODE(code)[i];
         int opcode = instr->op.code;
-        if (opcode == INSTRUMENTED_LINE) {
+        int oparg = instr->op.arg;
+        if (opcode == ENTER_EXECUTOR) {
+            _PyExecutorObject *exec = code->co_executors->executors[oparg];
+            opcode = exec->vm_data.opcode;
+            oparg = exec->vm_data.oparg;
+        }
+        else if (opcode == INSTRUMENTED_LINE) {
             opcode = code->_co_monitoring->lines[i].original_opcode;
         }
+        assert(opcode != ENTER_EXECUTOR);
```

Comment:
This assert should be moved before the original `if (opcode == INSTRUMENTED_LINE) {`, we shouldn't get here with and `ENTER_EXECUTOR` present.

Reply:
Move it to L1310 will be enough?